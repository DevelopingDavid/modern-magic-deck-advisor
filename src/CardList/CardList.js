import React from 'react'
import CardListItem from '../CardListItem/CardListItem.js';
import '../styles/main.scss';

function CardList(props) {
  const cardNames = props.userCardsData.map(card => card.cardName);
  cardNames.sort();
  return (
    <div>
      <h1>My Cards:</h1>
      <ul className="cardlist--ul">
        {
          cardNames.map((cardName, index) => {
            return (
              <CardListItem 
                cardName={cardName}
                cardIndex={index}
                cardCount={props.userCardsData[index].cardCount}
                key={cardName}
                userCardsData={props.userCardsData}
                saveArray={props.saveArray}/>
            )
          }, this)
        }
      </ul>
      <button onClick={props.compareBuilds}>Compare Builds</button>
    </div>
  )
}

export default CardList;