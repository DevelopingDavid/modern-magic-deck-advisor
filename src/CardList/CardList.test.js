import React from 'react';
import CardList from './CardList';
import { shallow } from 'enzyme';

const userCardsData = [
  {
    cardName: 'Blood Moon',
    cardCount: 2
  }
]
describe('CardList', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardList
        userCardsData={userCardsData}
       /> 
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})