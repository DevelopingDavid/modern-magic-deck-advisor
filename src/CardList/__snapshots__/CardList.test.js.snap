// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should match the snapshot 1`] = `
<div>
  <h1>
    My Cards:
  </h1>
  <ul
    className="cardlist--ul"
  >
    <CardListItem
      cardCount={2}
      cardIndex={0}
      cardName="Blood Moon"
      key="Blood Moon"
      userCardsData={
        Array [
          Object {
            "cardCount": 2,
            "cardName": "Blood Moon",
          },
        ]
      }
    />
  </ul>
  <button>
    Compare Builds
  </button>
</div>
`;
